packages/app-desktop/gui/Sidebar/Sidebar.tsx

```diff
@@ -428,6 +428,28 @@ const SidebarComponent = (props: Props) => {
 		menu.popup({ window: bridge().window() });
 	}, [props.folders, props.dispatch, pluginsRef]);
 
+	const toggleAllNotesContextMenu = useCallback(async (event: any) => {
+		const itemId = event.currentTarget.getAttribute('data-id');
+		if (itemId === Folder.conflictFolderId()) return;
+
+		const itemType = Number(event.currentTarget.getAttribute('data-type'));
+		if (!itemId || !itemType) throw new Error('No data on element');
+
+		const menu = new Menu();
+
+		if (itemType === BaseModel.TYPE_FOLDER) {
+			if (Setting.value('notes.perFolderSortOrderEnabled')) {
+				menu.append(new MenuItem({
+					...menuUtils.commandToStatefulMenuItem('togglePerFolderSortOrder', itemId),
+					type: 'checkbox',
+					checked: PerFolderSortOrderService.isSet(itemId),
+				}));
+			}
+		}
+
+		menu.popup({ window: bridge().window() });
+	}, []);
+
```

Comment:
Alright, I refactored the code and had to resolve some merge conflicts. The changes I made were only adding if-else statements and boolean expressions to existing if-elses.

Reply:
I see that `new MenuItem({ label: module.fullLabel(),` changed to `new MenuItem({label: deleteButtonLabel,`. Those I did not do, maybe those were from the merge conflict.

Reply:
Hmm, yes but adding if-elses everywhere, and just for that particular folder item is not really an improvement unfortunately. It significantly increases the complexity of the program and will make it difficult to make further changes to this function.

So the problem is that you copied and pasted a lot of code and put it there. How about going back to this but then simplify it. For example, you had a check on the conflict folder - why is that needed? It probably isn't since that function was only called for the "All notes" item.

Then you have some check on the item ID - again why? This function is called`toggleAllNotesContextMenu` so it only applies to "All note", so there's only one possible ID.

Once you removed all that not necessary, probably what's left is more manageable. It might be that a part of the code needs to be moved to a function and called from here, and from wherever you copied the code from.

Reply:
So using `onClick={() => { onAllNotesClick_; folderItem_click(all_notes_id); }` works but it changes the folder state twice so I had to add more conditions in here: 
```typescript
const allNotesSelected = props.notesParentType === 'SmartFilter' && props.selectedSmartFilterId === ALL_NOTES_FILTER_ID;
```
 so the `All Notes` stays highlighted after clicking.

I am thinking about another solution that uses only `onAllNotesClick_`.  and adds these changes in `PerFolderSortOrderService.ts`: 

```typescript
private static getSelectedFolderId(): string {
		if (this.folderState.notesParentType === 'Folder') {
			return this.folderState.selectedFolderId;
		} else if (this.folderState.notesParentType === 'SmartFilter') {
			return 'all_notes_folder_id';
		} else {
			return '';
		}
}
```
I think this is better because is cleaner and we do not change the folder state twice.